### PR TITLE
Transform com_user to ES6 

### PR DESCRIPTION
**Pull request by anuragteapot**
Pull Request for Issue #16

### CODE REVIEW

File: media/com_users/js/admin-users-groups.es6.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+((Joomla) => {
+  'use strict';
+
+  document.addEventListener('DOMContentLoaded', () => {
+    Joomla.submitbutton = (task) => {
+      if (task === 'groups.delete') {
+        const cids = document.getElementsByName('cid[]');
+        for (let i = 0; i < cids.length; i++) {
+          if (cids[i].checked && cids[i].parentNode.getAttribute('data-usercount') !== 0) {
+            if (window.confirm(Joomla.JText._('COM_USERS_GROUPS_CONFIRM_DELETE'))) {
```
[the remark this answers is not in the document]
Comment (dgrammatiko) [reply media/com_users/js/admin-users-groups.es6.js]: `const cids = [].slice.call(document.getElementsByName('cid[]'));`

then 

`cids.forEach((element) => { ... });`

Also please use meaningful variable names, cids is not clear what we're dealing with

**Comment by dneukirchen** [comment]
🐶 :D can you fix that?

**Comment by anuragteapot** [comment]
yes, wait :)

File: media/com_users/js/admin-users-groups.es6.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Joomla = window.Joomla || {};
+
+((Joomla) => {
+  'use strict';
+
+  document.addEventListener('DOMContentLoaded', () => {
+    Joomla.submitbutton = (task) => {
+      if (task === 'groups.delete') {
+        const cids = document.getElementsByName('cid[]');
+        for (let i = 0; i < cids.length; i++) {
```
[the remark this answers is not in the document]
Comment (dneukirchen) [reply media/com_users/js/admin-users-groups.es6.js]: We should probably allow this in loops? Its pretty handy. (https://eslint.org/docs/rules/no-plusplus) "allowForLoopAfterthoughts": true
What do you think?

[the remark this answers is not in the document]
Comment (dgrammatiko) [reply media/com_users/js/admin-users-groups.es6.js]: Personally I prefer `array.forEach()` as it's closer to functional programming, but I guess we're nowhere near that in our current codebase, so meh

**Comment by anuragteapot** [comment]
@dneukirchen what should i do for <code>Unexpected alert no-alert</code> ?

**Comment by dneukirchen** [comment]
unfortunately we only have Joomla.renderMessages() to replace the alerts.

**Comment by C-Lodder** [comment]
Ignore the alerts for now and we can create our own custom element for that

**Comment by anuragteapot** [comment]
Thanks :)

File: media/com_users/js/admin-users-groups.es6.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (anuragteapot) [reply media/com_users/js/admin-users-groups.es6.js]: Is <code>forEach</code> works fine with return ?

**Review by dneukirchen — CHANGES_REQUESTED** [review]
Add //Todos and we can merge it

**Comment by anuragteapot** [comment]
@dneukirchen done!

File: media/com_users/js/admin-users-groups.js [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
+
 /**
  * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
 Joomla = window.Joomla || {};
 
-(function(Joomla) {
-	'use strict';
```
Comment (anuragteapot) [inline media/com_users/js/admin-users-groups.js:9]: Please review at this point

Comment (anuragteapot) [reply media/com_users/js/admin-users-groups.js:9]: I removed Joomla parameter is this correct ?

File: media/com_users/js/admin-users-user.es6.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+(() => {
+  'use strict';
+
+  document.addEventListener('DOMContentLoaded', () => {
+    Joomla.twoFactorMethodChange = () => {
+      const method = document.getElementById('jform_twofactor_method');
+      if (method) {
+        const selectedPane = `com_users_twofactor_${method.value}`;
+        const twoFactorForms = document.querySelectorAll('#com_users_twofactor_forms_container > div');
+        twoFactorForms.forEach((value) => {
```
Comment (dneukirchen) [inline media/com_users/js/admin-users-user.es6.js]: nodeList.forEach is not supported in ie

**Review by dneukirchen — CHANGES_REQUESTED** [review]

File: media/com_users/js/admin-users-groups.js (hunk 3 above)
Comment (dneukirchen) [reply media/com_users/js/admin-users-groups.js:9]: Technically its not wrong to remove the parameter, but it has benefits to keep it: 
* uglify can minify it (smaller filesize)
* its clear what dependencies this script needs
* its easier to test

so i recommend (re-)adding all dependencies and pass them as params

**Pull request closed without merging**